src/pcap/bosh_resolver.go

```diff
@@ -0,0 +1,208 @@
+package pcap
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+
+	"github.com/cloudfoundry/pcap-release/src/pcap/bosh"
+)
+
+type BoshResolver struct {
+	environment bosh.Environment
+	client      *http.Client
+	uaaURLs     []string
+	agentPort   int
+}
+
+func NewBoshResolver(environment bosh.Environment, agentPort int) (*BoshResolver, error) {
+	resolver := &BoshResolver{environment: environment, agentPort: agentPort} // TODO: get agent port from where?
+	err := resolver.setup()
+	if err != nil {
+		return nil, err
+	}
+	return resolver, err
+}
+
+func (br *BoshResolver) Name() string {
+	// TODO we need to differentiate between bosh resolvers for different environments (i.e. bootstrap-bosh, bosh. This would take some refactoring, e.g. in pcap.proto StatusResponse
```

Comment:
So do you want to signal, which environments are supported on the server as part of the status message?

We could consider having a `resolvers` field that just contains the names (and types?) of all known resolvers, instead of having the hardcoded `cf` and `bosh` boolean flags.